### PR TITLE
Adds extra i/o variables for the aging process

### PR DESCRIPTION
**Pull request by singhbalwinder**
This is an old branch for some extra aging i/o. This branch also adds the capability to print 2d logical arrays.

### CODE REVIEW

**Review by tangshuaiqi — APPROVED** [review]
Looks good to me. Just note that the refactor-maint-2.0 already has my PR with `write_2d_var_int` added as well as `lbound/ubound` change. There may be a conflict need to be solved to merge this PR.

**Comment by singhbalwinder** [comment]
Yeah, I am aware of that. I will resolve it while merging or rebasing this.

File: components/eam/src/chemistry/utils/yaml_input_file_io.F90 [hunk 1]
```diff
@@ -1086,6 +1096,102 @@ subroutine write_2d_output_var(unit_output, fld_name, dim1, dim2, field, inp_out
...
+    character(len=*), intent(in) :: fld_name ! name of the field
+    integer, intent(in)  :: field(:,:)        ! field values (integer)
+
+    !local
+    integer :: d1, d2
+
+    !check if file is open to write or not
+    call is_file_open(unit_input)
+
+    !format statement to write in double precision
```
Comment (brigaudet) [inline components/eam/src/chemistry/utils/yaml_input_file_io.F90]: Maybe change the above comment?

**Review by brigaudet — APPROVED** [review]
Looks good, nothing to add, except for a minor comment / question.

**Comment by singhbalwinder** [comment]
$ /compyfs/sing201/e3sm_scratch/aging_vars/cs.status.comp_aging_vars_mam4_org_v2_baselines -s
comp_aging_vars_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**